Makefile

```diff
@@ -112,22 +112,31 @@ manifests: controller-gen kustomize ## Generates the manifests in $PROJECT_DIR/i
 	controller-gen crd:crdVersions=v1 rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=install/crd output:rbac:artifacts:config=install/rbac && $(KUSTOMIZE) build install > $(AUTHORINO_MANIFESTS)
 	$(MAKE) patch-webhook
 
+run:GIT_SHA=$(shell git rev-parse HEAD)
+run:DIRTY=$(shell $(PROJECT_DIR)/hack/check-git-dirty.sh || echo "unknown")
 run: generate manifests ## Runs the application against the Kubernetes cluster configured in ~/.kube/config
-	go run -ldflags "-X main.version=$(VERSION)" ./main.go server
+	go run -ldflags "-X main.version=v$(VERSION) -X main.gitSHA=${GIT_SHA} -X main.dirty=${DIRTY}" ./main.go server
 
+build:GIT_SHA=$(shell git rev-parse HEAD)
+build:DIRTY=$(shell $(PROJECT_DIR)/hack/check-git-dirty.sh || echo "unknown")
 build: generate ## Builds the manager binary
-	CGO_ENABLED=0 GO111MODULE=on go build -a -ldflags "-X main.version=$(VERSION)" -o bin/authorino main.go
+	CGO_ENABLED=0 GO111MODULE=on go build -a -ldflags "-X main.version=v$(VERSION) -X main.gitSHA=${GIT_SHA} -X main.dirty=${DIRTY}" -o bin/authorino main.go
 
 IMAGE_REPO ?= authorino
+VERSION ?= $(GIT_SHA)
 using_semantic_version := $(shell [[ $(VERSION) =~ ^[0-9]+\.[0-9]+\.[0-9]+(-.+)?$$ ]] && echo "true")
 ifdef using_semantic_version
 IMAGE_TAG=v$(VERSION)
 else
 IMAGE_TAG=local
 endif
 AUTHORINO_IMAGE ?= $(IMAGE_REPO):$(IMAGE_TAG)
+
+docker-build: AUTHORINO_VERSION=v0.17.3-dev # change as version increments
```

Comment:
This is going to be very easy to overlook and forget, especially at line 135. I wonder if we should read it from a file [`build.yaml`](https://github.com/Kuadrant/authorino-operator/blob/7e2fb1b5a7ce863d417cddcb3abb1083aba5a12f/Makefile#L87) file in the root of the project?

@didierofrivia @adam-cattermole, you both have experience with reading this kind of parameters for the release process. WDYT?

Reply:
I would tend to agree that it could be easy to forget this and is another extra manual maintenance step for releases. That being said, if I'm not mistaken, the make target isn't used by the release image build workflow so perhaps not the end of the world if it gets missed - although that does make me wonder what is the value in setting it explicitly like this for local dev only?

I don't mind the use of the config file (build.yaml) that we used in the operator where it was autogenerated on release branches to ensure consistency but I'm not as keen on one that might sit around on main indefinitely

Reply:
Maybe we could check the latest tag on the repo, increment by 1 and append `-dev` ? It would require a bit of string formatting but I think it could be possible.

Reply:
what about `version/version.go` like we have in other go projects? I wish this would be in go.mod like in rust. But since it is not in go.mod, version.version.go looks best approach to me.

Reply:
@eguzki, do you mean the static version in the go file? If so, no longer inject the value at build time?

Reply:
I believe so, here is an example of that in effect  https://github.com/Kuadrant/kuadrantctl/blob/main/version/version.go .

Reply:
We can't do that, I'm afraid. It has implication downstream as well.

Reply:
OK sure, I will stick to the above plan, of implementing a `build.yaml` file at project root with the version there...

Something like:

```yaml
# Build params for compiling build image . 
build:
   version: <version>
   ... : ...
   ...
```

Reply:
We won't change source code files downstream, @eguzki. I prefer not to touch this on a component that is actively being productised. The version number is merely informative and not worth the trouble.

Reply:
up to you, obviously. But this is not a blocking thing at all. Changing things on downstream is regular work being done daily basis. 

This effort is on pushing to have consistency across all golang components in kuadrant.

Reply:
Thanks, @eguzki. Understood.

Let's please not hard code the version number of Authorino inside a Golang file.

Personally, I would prefer this to be the policy setting consistency across all Kuadrant components, but I understand it can be tricky for some; the ones written in Rust in particular. Either way, let's take the discussion outside of this PR, so it can progress.

For now, I vote moving forward with your [proposed](https://github.com/Kuadrant/authorino/pull/473#discussion_r1692951848) `build.yaml`, @ehearneRedHat.